core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/RemoveRedundantTableScanPredicate.java

```diff
@@ -137,20 +111,25 @@ private PlanNode removeRedundantTableScanPredicate(
             return predicateColumnDomain.intersect(enforcedColumnDomain);
         });
 
+        if (unenforcedDomain.equals(predicateDomain)) {
+            // no change in filter predicate
+            return Result.empty();
+        }
+
         Map<ColumnHandle, Symbol> assignments = ImmutableBiMap.copyOf(node.getAssignments()).inverse();
         Expression resultingPredicate = createResultingPredicate(
                 plannerContext,
-                session,
-                symbolAllocator,
+                context.getSession(),
+                context.getSymbolAllocator(),
                 typeAnalyzer,
-                new DomainTranslator(plannerContext).toPredicate(session, unenforcedDomain.transformKeys(assignments::get)),
+                new DomainTranslator(plannerContext).toPredicate(context.getSession(), unenforcedDomain.transformKeys(assignments::get)),
```

Comment:
nit: assigning `context.getSession()` to a variable would make for a smaller diff and arguably more readable code